experiments/sample_ssl4eo.py

```diff
@@ -0,0 +1,193 @@
+    )
+    return bbox
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--save-path", type=str, default="./data/", help="dir to save data"
+    )
+    parser.add_argument(
+        "--radius", type=float, default=1320, help="patch radius in meters"
+    )
+    parser.add_argument(
+        "--overlap-ratio",
+        type=float,
+        default=0.25,
+        help="max overlap ratio between adjacent patches",
+    )
+    parser.add_argument(
+        "--num-cities", type=int, default=10000, help="number of cities to sample"
+    )
+    parser.add_argument(
+        "--std", type=int, default=50, help="std of gaussian distribution"
+    )
+    parser.add_argument(
+        "--resume", type=str, default=None, help="resume from a previous run"
+    )
+    parser.add_argument(
+        "--indices-range",
+        type=int,
+        nargs=2,
+        default=[0, 250000],
+        help="indices to download",
+    )
+
+    args = parser.parse_args()
+    np.random.seed(42)
+
+    # if resume
+    ext_coords = {}
+    if args.resume:
+        ext_path = args.resume
+        with open(ext_path) as csv_file:
+            reader = csv.reader(csv_file)
+            for row in reader:
+                key = row[0]
+                val1 = float(row[1])
+                val2 = float(row[2])
+                ext_coords[key] = (val1, val2)  # lon, lat
+    else:
+        ext_path = os.path.join(args.save_path, "sampled_locations.csv")
+
+    # initialize sampler
+    cities = get_world_cities()
+    interest_points = get_interest_points(cities, size=args.num_cities)
+    bbox_size = (
+        (1 - args.overlap_ratio) * args.radius / 1000
+    )  # allow little overlap between adjacent patches
+    bbox_size_degree = km2deg(bbox_size)
+
+    # build rtree
+    rtree_coords = index.Index()
+    if args.resume:
+        print("Load existing locations.")
+        for i, key in enumerate(tqdm(ext_coords.keys())):
+            c = ext_coords[key]
+            bbox = create_bbox(c, bbox_size_degree)
+            rtree_coords.insert(i, bbox)
+
+    # sample locations
+    start_time = time.time()
+    indices = range(args.indices_range[0], args.indices_range[1])
+
+    for idx in tqdm(indices):
+        # skip if already sampled
+        if str(idx) in ext_coords.keys():
+            if args.resume:
+                continue
+
+        # sample new coord and check overlap
+        count = 0
+        while count < 1:
+            new_coord = sample_point(
+                interest_points, args.std
+            )  # (lon,lat) of top-10000 cities
+            bbox = create_bbox(new_coord, bbox_size_degree)
+            if list(rtree_coords.intersection(bbox)):
+                continue
+            rtree_coords.insert(idx, bbox)
+            count += 1
+
+        # save to file
+        with open(ext_path, "a") as f:
+            writer = csv.writer(f)
+            data = [idx, new_coord[0], new_coord[1]]
```

Comment:
```suggestion
            data = [idx] + new_coord
```

Reply:
new_coord is a tuple (lon,lat), so maybe keep it?

Reply:
In that case you can use:
```suggestion
            data = [idx, *new_coord]
```